File: scripts/create_admin_user.py
#!/usr/bin/env python3
"""
Script para criar usuário admin inicial no banco de dados.
Uso: python3 scripts/create_admin_user.py
"""
import sys
import os

# Adicionar o diretório raiz ao path
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))

from api.database import SessionLocal
from api import models
from api.auth import get_password_hash


def create_admin_user():
    """Cria usuário admin se não existir"""
    db = SessionLocal()

    try:
        # Verificar se já existe usuário admin
        existing_user = db.query(models.User).filter(
            models.User.username == 'admin'
        ).first()

        if existing_user:
            print("✓ Usuário 'admin' já existe!")
            return

        # Criar novo usuário admin
        admin_user = models.User(
            email='admin@sceap.com',
            username='admin',
            hashed_password=get_password_hash('admin123'),
            is_active=True,
            is_superuser=True
        )

        db.add(admin_user)
        db.commit()
        db.refresh(admin_user)

        print("✓ Usuário admin criado com sucesso!")
        print(f"  Email: {admin_user.email}")
        print(f"  Username: {admin_user.username}")
        print(f"  Password: admin123")
        print("\n⚠️  IMPORTANTE: Altere a senha após o primeiro login!")

    except Exception as e:
        db.rollback()
        print(f"✗ Erro ao criar usuário: {e}")
        sys.exit(1)
    finally:
        db.close()


if __name__ == '__main__':
    create_admin_user()

File: scripts/create-admin-user.js
#!/usr/bin/env node
/**
 * CREATE ADMIN USER
 * Cria usuário admin padrão se não existir nenhum usuário
 *
 * Uso: npm run create-admin
 * Ou: node scripts/create-admin-user.js
 */

import pg from 'pg';
import bcrypt from 'bcryptjs';
import dotenv from 'dotenv';

dotenv.config();

async function createAdminUser() {
  console.log('═'.repeat(70));
  console.log('👤 CRIAR USUÁRIO ADMIN');
  console.log('═'.repeat(70));
  console.log('');

  if (!process.env.DATABASE_URL) {
    console.log('❌ DATABASE_URL não configurado!');
    process.exit(1);
  }

  const client = new pg.Client({
    connectionString: process.env.DATABASE_URL,
    ssl: process.env.NODE_ENV === 'production'
      ? { rejectUnauthorized: false }
      : false
  });

  try {
    await client.connect();
    console.log('✅ Conectado ao PostgreSQL');
    console.log('');

    // Verificar se já existem usuários
    const usersCount = await client.query('SELECT COUNT(*) FROM users');
    const totalUsers = parseInt(usersCount.rows[0].count);

    console.log(`📊 Total de usuários no banco: ${totalUsers}`);
    console.log('');

    if (totalUsers > 0) {
      // Listar usuários existentes
      const users = await client.query(`
        SELECT email, name, role, created_at
        FROM users
        ORDER BY created_at DESC
      `);

      console.log('👥 USUÁRIOS EXISTENTES:');
      console.log('─'.repeat(70));
      users.rows.forEach((user, idx) => {
        console.log(`${idx + 1}. ${user.email}`);
        console.log(`   Nome: ${user.name || 'N/A'}`);
        console.log(`   Role: ${user.role}`);
        console.log(`   Criado: ${new Date(user.created_at).toLocaleString('pt-BR')}`);
        console.log('');
      });

      console.log('⚠️  Usuários já existem. Não criando admin padrão.');
      console.log('');
      console.log('Para fazer login use um dos emails acima.');
      console.log('');
    } else {
      // Criar usuário admin padrão
      console.log('📝 Nenhum usuário encontrado. Criando admin padrão...');
      console.log('');

      const defaultAdmin = {
        email: 'admin@iarom.com.br',
        password: 'Admin@123',
        name: 'Administrador ROM',
        role: 'admin',
        oab: null
      };

      const passwordHash = await bcrypt.hash(defaultAdmin.password, 10);

      await client.query(`
        INSERT INTO users (email, password_hash, name, role, oab)
        VALUES ($1, $2, $3, $4, $5)
      `, [
        defaultAdmin.email,
        passwordHash,
        defaultAdmin.name,
        defaultAdmin.role,
        defaultAdmin.oab
      ]);

      console.log('✅ ADMIN CRIADO COM SUCESSO!');
      console.log('');
      console.log('━'.repeat(70));
      console.log('📋 CREDENCIAIS DE ACESSO:');
      console.log('━'.repeat(70));
      console.log(`Email:    ${defaultAdmin.email}`);
      console.log(`Senha:    ${defaultAdmin.password}`);
      console.log('━'.repeat(70));
      console.log('');
      console.log('⚠️  IMPORTANTE: Altere a senha após o primeiro login!');
      console.log('');
    }

  } catch (error) {
    console.error('');
    console.error('❌ ERRO:', error.message);
    console.error('');
    process.exit(1);
  } finally {
    await client.end();
  }

  console.log('═'.repeat(70));
  console.log('');
}

createAdminUser().catch(error => {
  console.error('');
  console.error('💥 ERRO FATAL');
  console.error(error);
  console.error('');
  process.exit(1);
});
